lib/framework/elastic_connector_actions.rb

```diff
@@ -12,88 +12,137 @@
-    end
+        client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
+        Utility::Logger.info("Successfully updated configuration for connector #{connector_package_id}")
+      end
 
-    def self.claim_job(connector_package_id)
-      body = {
-        :doc => {
-          :sync_now => false,
-          :sync_status => Connectors::SyncStatus::IN_PROGRESS,
-          :last_synced => Time.now
+      def claim_job(connector_package_id)
+        body = {
+          :doc => {
+            :sync_now => false,
+            :sync_status => Connectors::SyncStatus::IN_PROGRESS,
+            :last_synced => Time.now
+          }
         }
-      }
 
-      client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
-      Utility::Logger.info("Successfully claimed job for connector #{connector_package_id}")
-    end
+        client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
+        Utility::Logger.info("Successfully claimed job for connector #{connector_package_id}")
+      end
 
-    def self.complete_sync(connector_package_id, error)
-      body = {
-        :doc => {
-          :sync_status => error.nil? ? Connectors::SyncStatus::COMPLETED : Connectors::SyncStatus::FAILED,
-          :sync_error => error,
-          :last_synced => Time.now
+      def complete_sync(connector_package_id, error)
+        body = {
+          :doc => {
+            :sync_status => error.nil? ? Connectors::SyncStatus::COMPLETED : Connectors::SyncStatus::FAILED,
+            :sync_error => error,
+            :last_synced => Time.now
+          }
         }
-      }
 
-      client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
+        client.update(:index => CONNECTORS_INDEX, :id => connector_package_id, :body => body)
 
-      if error
-        Utility::Logger.info("Failed to sync for connector #{connector_package_id} with error #{error}")
-      else
-        Utility::Logger.info("Successfully synced for connector #{connector_package_id}")
+        if error
+          Utility::Logger.info("Failed to sync for connector #{connector_package_id} with error #{error}")
+        else
+          Utility::Logger.info("Successfully synced for connector #{connector_package_id}")
+        end
       end
-    end
 
-    def self.client
-      @client ||= Utility::EsClient.new
-    end
+      def client
+        @client ||= Utility::EsClient.new
+      end
 
-    # should only be used in CLI
-    def self.ensure_index_exists(index_name)
-      client.indices.create(:index => index_name) unless client.indices.exists?(:index => index_name)
-    end
+      # should only be used in CLI
+      def ensure_index_exists(index_name, body = {})
+        client.indices.create(:index => index_name, :body => body) unless client.indices.exists?(:index => index_name)
```

Comment:
Should this check to make sure the response is a 20x? Otherwise `ensure_index_exists` isn't really ensuring anything, it's just attempting to create it.

Reply:
If it's not 20x, I think it will just raise an error, and the app/CLI will exit. We should probably catch it and display more user-friendly message. cc @artem-shelkovnikov @mchernyavskaya

Reply:
@artem-shelkovnikov, the connector service doesn't handle edge cases well. I think it's fine as we want to move fast. We can have a dedicated PR later to make the connector service more robust.

Reply:
Good call, 100% agree

Reply:
For when you get to that client augmentation, I suggest looking at [this code](https://github.com/elastic/ent-search/blob/efeb730e1dda78631b7b8a47ed8ecc35c917f239/lib/swiftype/es/client.rb#L14-L30). I'm currently leveraging this Elasticsearch Client/wrapper for the Elasticsearch Crawler, and really liking how it works. I think it might be a good candidate for us to open so that we don't have to reinvent it from scratch.